System/ContentFile.h

```diff
@@ -32,483 +19,253 @@ struct ga_Handle;
-    static std::map<std::string, BITMAP *> m_sLoadedBitmaps[BitDepthCount];
-
-	static std::map<size_t, std::string> m_PathHashes;
-
-	// Static map containing all the already loaded FSOUND_SAMPLE:s and their paths
-    static std::map<std::string, AUDIO_STRUCT *> m_sLoadedSamples;
-
-    // Path to this ContentFile's Datafile Object's path. "datafile.dat#objectname"
-    // In the case of an animation, this filename/name will be appended with 000, 001, 002 etc
-    std::string m_DataPath;
-
-    // Data Module ID of where this was loaded from
-    int m_DataModuleID;
-
-    // Whether the data itself has been modified since load, and should be saved out again to the path on write
-    bool m_DataModified;
-
-    // Non-ownership pointer to the loaded data for convenience. Do not release/delete this.
-    void *m_pLoadedData;
-    // Size of loaded data
-    long m_LoadedDataSize;
-
-    // This is only if the data is loaded from a datafile;
-    // needs to be saved so that it can be unloaded as some point
-    DATAFILE *m_pDataFile;
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Private member variable and method declarations
-
-private:
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Clear
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Clears all the member variables of this ContentFile, effectively
-//                  resetting the members of this abstraction level only.
-// Arguments:       None.
-// Return value:    None.
-
-    void Clear();
-
-};
-
-} // namespace RTE
-
-#endif // File
\ No newline at end of file
+	/// <summary>
+	/// A representation of a content file that is stored either directly on disk or packaged in another file.
+	/// </summary>
+	class ContentFile : public Serializable {
+
+	public:
+
+#pragma region Creation
+		/// <summary>
+		/// Constructor method used to instantiate a ContentFile object in system memory. Create() should be called before using the object.
+		/// </summary>
+		ContentFile() { Clear(); }
+
+		/// <summary>
+		/// Constructor method used to instantiate a ContentFile object in system memory, and also do a Create() in the same line.
+		/// Create() should therefore not be called after using this constructor.
+		/// </summary>
+		/// <param name="filePath">A string defining the path to where the content file itself is located, either within the package file, or directly on the disk.</param>
+		ContentFile(const char *filePath) { Clear(); Create(filePath); }
+
+		/// <summary>
+		/// Constructor method used to instantiate a ContentFile object in system memory from a hash value of the file path, and also do a Create() in the same line.
+		/// Create() should therefore not be called after using this constructor.
```

Comment:
I'm not sure we need this note - if a constructor does Create it's probably obvious you shouldn't do it again.